management/server/store.go

```diff
@@ -76,50 +79,74 @@ func getStoreEngineFromEnv() StoreEngine {
 	}
 
 	value := StoreEngine(strings.ToLower(kind))
-	if value == FileStoreEngine || value == SqliteStoreEngine || value == PostgresStoreEngine {
+	if value == SqliteStoreEngine || value == PostgresStoreEngine {
 		return value
 	}
 
 	return SqliteStoreEngine
 }
 
-func getStoreEngineFromDatadir(dataDir string) StoreEngine {
-	storeFile := filepath.Join(dataDir, storeFileName)
-	if _, err := os.Stat(storeFile); err != nil {
-		// json file not found then use sqlite as default
-		return SqliteStoreEngine
-	}
-	return FileStoreEngine
-}
-
-func NewStore(kind StoreEngine, dataDir string, metrics telemetry.AppMetrics) (Store, error) {
+// getStoreEngine determines the store engine to use
+func getStoreEngine(kind StoreEngine) StoreEngine {
 	if kind == "" {
-		// if store engine is not set in the config we first try to evaluate NETBIRD_STORE_ENGINE
 		kind = getStoreEngineFromEnv()
 		if kind == "" {
-			// NETBIRD_STORE_ENGINE is not set we evaluate default based on dataDir
-			kind = getStoreEngineFromDatadir(dataDir)
+			kind = SqliteStoreEngine
 		}
 	}
+	return kind
+}
+
+// NewStore creates a new store based on the provided engine type, data directory, and telemetry metrics
+func NewStore(kind StoreEngine, dataDir string, metrics telemetry.AppMetrics) (Store, error) {
+	kind = getStoreEngine(kind)
+
+	if err := checkFileStoreEngine(kind, dataDir); err != nil {
+		return nil, err
+	}
+
```

Comment:
It is not clear the `case handleUnsupportedStoreEngine` and the `checkFileStoreEngine`
The `handleUnsupportedStoreEngine` tries to migrate the json file to SQL. But before that, the checkFileStoreEngine will return with error because the JSON file exists.

Reply:
`checkFileStoreEngine` only checks if  `store.json` exists and  `jsonfile` is set as the store engine in `management.json`. It then returns an error with a link pointing to the steps for migrating from jsonfile to SQLite.

`handleUnsupportedStoreEngine` handles the case when the store engine is not supported. If there is an existing `store.json`, it automatically migrates to SQLite, but the user will need to update the store engine to SQLite. If there is no existing store.json, it returns an error indicating an unsupported store engine